doc/blog/Wasm-JS series/Blog - WASM.md

```diff
@@ -0,0 +1,672 @@
+
+This article will first review how Uno interoperates with HTML5, following by an integration of a simple JavaScript-based syntax highlighter in a Uno project. Another article will go further and deeper in the dotnet/JavaScript interop.
+
+# Uno Wasm Bootstrap - where it starts
+
+Behind a Uno-Wasm project, there's a component called [`Uno.Wasm.Bootstrap`](https://github.com/unoplatform/Uno.Wasm.Bootstrap). It contains the tooling required to build, package, deploy and run a _.NET_ project using _Mono_ in a web browser using _WebAssembly_. It's automatically include in any
+
+## Embedding assets
+
+In the HTML world, everything running in the browser must be downloaded from a server. This is called assets. To integrate existing javascript frameworks, you can either download them directly from the Internet (usually from a CDN service) or embed them with your app.
+
+Uno Wasm Bootstrap can automatically embed any asset and deploy them with the app. Some of them (CSS & JavaScript) could also be loaded with the app. Here's how to declare them in a _Uno Wasm_ project:
+
+1. **JavaScript files** should be in `WasmScripts`  folder: they will be copied to output folder and loaded automatically by the bootstrapper when the page loads. **They must be marked as `EmbeddedResources`**.
+
+2. **CSS Style files** should be in the `WasmCSS` folder: they will be copied to output folder and referenced in the _html head_ of the application. **They must be marked as `EmbeddedResources`**.
+
+3. **Asset files** should be marked as `Content` in the app. The file will be copied to output folder and will preserve the same relative path.
+
+4. Alternatively, **any kind of asset files** can be placed directly in the `wwwroot` folder as you would do with any standard _aspnet core_ project. They will be deployed with the app, but the application code will have the responsibility to fetch and use them.
+
+   > **Is it an Aspnet Core project?**
+   > Nope. Some of the deployment features, like the `wwwroot` folder, the VisualStudio integration for running and debugging is reused as an Aspnet Core project, but that's it. The C# code you put in such project will run in the browser, using the mono-wasm runtime. Right now there's no _server side_ component in such project.
+
+## Uno-Wasm controls are actually HTML5 elements
+
+The [philosophy of Uno](https://platform.uno/docs/articles/concepts/overview/philosophy-of-uno.html) is to rely on native platforms as much as possible. In the context of a browser, that's the HTML5 DOM. It means each time you're creating an instance of a class deriving from `UIElement`, you're actually creating a HTML element.
+
+That aslo means you can control how this element is created.  By default is a `<div>`, but you can change this directly in the constructor by setting the `htmlTag` parameter to whatever required. Example:
```

Comment:
```suggestion
That also means that it is possible to control how this element is created.  By default it is a `<div>`, but it can be changed in the constructor by providing the `htmlTag` parameter to the one required. For example:
```